src/lib/get-request-options-with-api-key.ts

```diff
@@ -0,0 +1,30 @@
+import { AxiosRequestConfig, AxiosRequestHeaders } from 'axios';
+
+import { Configuration } from '../configuration';
+
+export const getRequestOptionsWithApiKey = async (
+  method: 'GET' | 'POST' | 'DELETE' | 'PUT' | 'PATCH',
+  configuration: Configuration,
+  options: AxiosRequestConfig = {}
+): Promise<AxiosRequestConfig> => {
+  const baseOptions = configuration.baseOptions;
+
+  const requestHeaderParameter = {} as AxiosRequestHeaders;
+
+  const requestOptions = { method, ...baseOptions, ...options };
+
+  if (configuration?.apiKey) {
+    const requestApiKeyValue =
+      typeof configuration.apiKey === 'function'
+        ? await configuration.apiKey('Authorization')
```

Comment:
Why would this be a function?

Reply:
Probably had way too much freedom in the definition of apiKey in Configuration. I played around a little bit with [openapi-generator](https://github.com/OpenAPITools/openapi-generator) and it would always allow config settings to be defined like this. I liked the idea of allowing promises, and tbh I just thought that if they did this, then there must be someone out there that wants to pass in a key via a function.
```
  apiKey: string | Promise<string> | ((name: string) => string) | ((name: string) => Promise<string>);
```
I changed to this to still allow promises if someone wanted to do that, but the function weirdness.
```
apiKey: string | Promise<string>
```

The code above simplifies to `const requestApiKeyValue = await configuration.apiKey`

Reply:
I probably included way too much freedom in the definition of apiKey in Configuration. I played around a little bit with [openapi-generator](https://github.com/OpenAPITools/openapi-generator) and it would always allow config settings to be defined like this. I liked the idea of allowing promises, and tbh I just thought that if they did this, then there must be someone out there that wants to pass in a key via a function.
```
  apiKey: string | Promise<string> | ((name: string) => string) | ((name: string) => Promise<string>);
```
I changed to this to still allow promises if someone wanted to do that, but the function weirdness.
```
apiKey: string | Promise<string>
```

The code above simplifies to `const requestApiKeyValue = await configuration.apiKey`